bin/wasm-node/javascript/src/worker/bindings-smoldot-light.ts

```diff
@@ -0,0 +1,310 @@
+// Smoldot
+// Copyright (C) 2019-2022  Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Exports a function that provides bindings for the bindings found in the Rust part of the code.
+//!
+//! In order to use this code, call the function passing an object, then fill the `instance` field
+//! of that object with the Wasm instance.
+
+import { Buffer } from 'buffer';
+import Websocket from 'websocket';
+import * as compat from '../compat/index.js';
+import type { SmoldotWasmInstance } from './bindings.js';
+
+import type { Socket as TcpSocket } from 'net';
+
+export interface Config {
+    instance?: SmoldotWasmInstance,
```

Comment:
why is it optional?

Reply:
We need to pass the bindings as parameter to `WebAssembly.instantiate`, so there's absolutely no way to pass an `instance` here. The way the code works is we create the bindings, then add the `instance` to the config after it is created.